packages/fxa-content-server/app/scripts/models/reliers/relier.js

```diff
@@ -161,6 +164,28 @@ var Relier = BaseRelier.extend({
     });
   },
 
+  fetchSubscriptionProductName() {
+    const subscriptionProductId = this.get('subscriptionProductId');
+    if (!subscriptionProductId) {
+      return Promise.resolve();
+    }
+    const subscriptionProductName = this.get('subscriptionProductName');
+    if (subscriptionProductName) {
+      return Promise.resolve();
+    }
+    const productNameUrl = `${this._config.authServerUrl}/v1/oauth/subscriptions/productname?productId=${subscriptionProductId}`;
+    return this.window
+      .fetch(productNameUrl)
```

Comment:
nit: this probably should use https://github.com/mozilla/fxa/blob/main/packages/fxa-content-server/app/scripts/lib/xhr.js for consistency.

Reply:
Oh hey, I hadn't even seen that library!